File: app/src/main/java/kg/koronastaff/staffapp/adapters/FAQAdapter.java
package kg.koronastaff.staffapp.adapters;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;

import kg.koronastaff.staffapp.R;
import kg.koronastaff.staffapp.models.FAQ;
import kg.koronastaff.staffapp.models.StationMap;

public class FAQAdapter extends RecyclerView.Adapter<FAQAdapter.MyViewHolder> {
    private ArrayList<FAQ> mDataset;

    static class MyViewHolder extends RecyclerView.ViewHolder {
        CardView frame;
        MyViewHolder(CardView v) {
            super(v);
            frame = v;
        }
    }

    public FAQAdapter(ArrayList<FAQ> myDataset) {
        mDataset = myDataset;
    }

    @Override
    public FAQAdapter.MyViewHolder onCreateViewHolder(
            ViewGroup parent, int viewType) {

        CardView v = (CardView) LayoutInflater.from(parent.getContext())
                .inflate(R.layout.quarantine_steps_item, parent, false);

        MyViewHolder vh = new MyViewHolder(v);
        return vh;
    }

    @Override
    public void onBindViewHolder(MyViewHolder holder, int position) {
        FAQ s = this.mDataset.get(position);
        TextView question = holder.frame.findViewById(R.id.step_num);
        TextView answer = holder.frame.findViewById(R.id.step_body);
        String b = s.getAnswer();
        if (b.length() > 199) {
            b = b.substring(0, 200).concat("...");
        }
        answer.setText(b);
        question.setText(s.getQuestion());
    }

    @Override
    public int getItemCount() {
        return mDataset.size();
    }

    public void update(ArrayList<FAQ> myDataset) {
        mDataset = myDataset;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/kg/koronastaff/staffapp/adapters/FAQAdapter.kt
package kg.koronastaff.staffapp.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import kg.koronastaff.staffapp.R
import kg.koronastaff.staffapp.models.FAQ
import kotlinx.android.synthetic.main.quarantine_steps_item.view.*
import java.util.*

class FAQAdapter(private var mDataset: ArrayList<FAQ>) : RecyclerView.Adapter<FAQAdapter.MyViewHolder>() {

    class MyViewHolder(var frame: CardView) : RecyclerView.ViewHolder(frame)

    override fun onCreateViewHolder(
            parent: ViewGroup, viewType: Int): MyViewHolder {
        val v = LayoutInflater.from(parent.context)
                .inflate(R.layout.quarantine_steps_item, parent, false) as CardView
        return MyViewHolder(v)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val s = mDataset[position]
        holder.frame.step_body.text = s.answer
        holder.frame.step_num.text = s.question
    }

    override fun getItemCount(): Int {
        return mDataset.size
    }

    fun update(myDataset: ArrayList<FAQ>) {
        mDataset = myDataset
        notifyDataSetChanged()
    }

}